packages/serverpod_serialization/test/ecode_with_type_protocol_test.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:serverpod_serialization/serverpod_serialization.dart';
+import 'package:test/test.dart';
+
+class _User implements SerializableModel, ProtocolSerialization {
+  final String name;
+  final String password;
+
+  _User({
+    required this.name,
+    required this.password,
+  });
+
+  @override
+  Map<String, String> toJson() {
+    return {
+      'name': name,
+      'password': password,
+    };
+  }
+
+  @override
+  Map<String, String> toJsonForProtocol() {
+    return {
+      'name': name,
+    };
+  }
+}
+
+class _TestProtocol extends SerializationManager {
+  @override
+  String? getClassNameForObject(Object? data) {
+    if (data is _User) return '_User';
+    if (data is List<_User>) return 'List<_User>';
+    if (data is List<Map<String, Object>>) return 'List<Map<String, Object>>';
```

Comment:
Question: Why object instead of `User? Same question for the line two rows down.